plugins/plugin-yaml/che-plugin-yaml-server/src/main/java/org/eclipse/che/plugin/yaml/server/languageserver/YamlLanguageServerConfig.java

```diff
@@ -47,75 +46,49 @@
+public class YamlLanguageServerConfig implements LanguageServerConfig {
 
+  private static final Logger LOG = LoggerFactory.getLogger(YamlLanguageServerConfig.class);
   private static final String REGEX = ".*\\.(yaml|yml)";
-  private static final LanguageServerDescription DESCRIPTION = createServerDescription();
+
   private static LanguageServer yamlLanguageServer;
+
   private final Path launchScript;
-  private HttpJsonRequestFactory requestFactory;
-  private String apiUrl;
+  private final EventService eventService;
+  private final HttpJsonRequestFactory requestFactory;
+  private final String apiUrl;
 
   @Inject
-  public YamlLanguageServerLauncher(
-      @Named("che.api") String apiUrl, HttpJsonRequestFactory requestFactory) {
+  public YamlLanguageServerConfig(
+      EventService eventService,
+      @Named("che.api") String apiUrl,
+      HttpJsonRequestFactory requestFactory) {
+    this.eventService = eventService;
     this.apiUrl = apiUrl;
     this.requestFactory = requestFactory;
     launchScript = Paths.get(System.getenv("HOME"), "che/ls-yaml/launch.sh");
   }
 
-  @Override
-  public boolean isAbleToLaunch() {
-    return Files.exists(launchScript);
-  }
-
-  protected LanguageServer connectToLanguageServer(
-      final Process languageServerProcess, LanguageClient client) {
-    Launcher<LanguageServer> launcher =
-        Launcher.createLauncher(
-            client,
-            LanguageServer.class,
-            languageServerProcess.getInputStream(),
-            languageServerProcess.getOutputStream());
-    launcher.startListening();
-    setYamlLanguageServer(launcher.getRemoteProxy());
-    return launcher.getRemoteProxy();
-  }
-
-  protected Process startLanguageServerProcess(String projectPath) throws LanguageServerException {
-    ProcessBuilder processBuilder = new ProcessBuilder(launchScript.toString());
-    processBuilder.redirectInput(ProcessBuilder.Redirect.PIPE);
-    processBuilder.redirectOutput(ProcessBuilder.Redirect.PIPE);
-    try {
-      return processBuilder.start();
-    } catch (IOException e) {
-      throw new LanguageServerException("Can't start YAML language server", e);
-    }
+  @PostConstruct
+  private void subscribe() {
+    eventService.subscribe(this::onLSPRoxyInitialized, LanguageServerInitializedEvent.class);
   }
 
   protected static LanguageServer getYamlLanguageServer() {
     return yamlLanguageServer;
   }
 
   protected static void setYamlLanguageServer(LanguageServer yamlLanguageServer) {
-    YamlLanguageServerLauncher.yamlLanguageServer = yamlLanguageServer;
+    YamlLanguageServerConfig.yamlLanguageServer = yamlLanguageServer;
   }
 
-  @Override
-  public void onServerInitialized(
-      LanguageServerLauncher launcher,
-      LanguageServer server,
-      ServerCapabilities capabilities,
-      String projectPath) {
+  public void onLSPRoxyInitialized(LanguageServerInitializedEvent event) {
```

Comment:
If this module uses a sidecar, why do we have a launch script? If not, why do we have a language proxy?

Reply:
If I understood correctly.

In fact `org.eclipse.lsp4j.services.LanguageServer` fits the definition of [proxy](https://en.oxforddictionaries.com/definition/proxy) quite well , because it acts as an intermediary for requests from a client to actual language server. And I'm sure you know (don't you?) that we anyway use it for both parallel container or installer + launcher, so it does not really matter if we have a launch script or not.

However I agree, to avoid further confusion I will make method name a bit clearer.